Fluid/TemplateOptions.cs

```diff
@@ -1,93 +1,111 @@
-
-        /// <summary>
-        /// Gets or sets the <see cref="CultureInfo"/> instance used to render locale values like dates and numbers.
-        /// </summary>
-        public CultureInfo CultureInfo { get; set; } = CultureInfo.InvariantCulture;
-
-        /// <summary>
-        /// Gets or sets the value returned by the "now" keyword.
-        /// </summary>
-        public Func<DateTimeOffset> Now { get; set; } = static () => DateTimeOffset.Now;
-
-        /// <summary>
-        /// Gets or sets the local time zone used when parsing or creating dates without specific ones.
-        /// </summary>
-        public TimeZoneInfo TimeZone { get; set; } = TimeZoneInfo.Local;
-
-        /// <summary>
-        /// Gets or sets the maximum depth of recursions a script can execute. 100 by default.
-        /// </summary>
-        public int MaxRecursion { get; set; } = 100;
-
-        /// <summary>
-        /// Gets the collection of filters available in the templates.
-        /// </summary>
-        public FilterCollection Filters { get; } = new FilterCollection();
-
-        /// <summary>
-        /// Gets a scope that is available in all the templates.
-        /// </summary>
-        public Scope Scope { get; } = new Scope();
-
-        /// <summary>
-        /// Gets the list of value converters.
-        /// </summary>
-        public List<Func<object, object>> ValueConverters { get; } = new List<Func<object, object>>();
-
-        /// <summary>
-        /// Gets or sets the delegate to execute when a Capture tag has been evaluated.
-        /// </summary>
-        public Func<string, string, ValueTask<string>> Captured { get; set; }
-
-        /// <summary>
-        /// Gets or sets the default trimming rules.
-        /// </summary>
-        public TrimmingFlags Trimming { get; set; } = TrimmingFlags.None;
-
-        /// <summary>
-        /// Gets or sets whether trimming is greedy. Default is true. When set to true, all successive blank chars are trimmed.
-        /// </summary>
-        public bool Greedy { get; set; } = true;
-
-
-        public TemplateOptions()
-        {
-            Filters.WithArrayFilters()
-                .WithStringFilters()
-                .WithNumberFilters()
-                .WithMiscFilters();
-        }
-    }
-}
+using Fluid.Filters;
+using Fluid.Values;
+using Microsoft.Extensions.FileProviders;
+using System.Globalization;
+
+namespace Fluid
+{
+    public class TemplateOptions
+    {
+        /// <param name="identifier">The name of the property that is assigned.</param>
+        /// <param name="value">The value that is assigned.</param>
+        /// <param name="context">The <see cref="TemplateContext" /> instance used for rendering the template.</param>
+        /// <returns>The value which should be assigned to the property.</returns>
+        public delegate ValueTask<FluidValue> AssignedDelegate(string identifier, FluidValue value, TemplateContext context);

```

Comment:
I'm not sure why you split into two delegates and why the signature is the same

Reply:
Honestly, I don't quite get this either. Whether you use assign or capture, it the same basic idea, conceptually.